packages/flutter_tools/test/hot_test.dart

```diff
@@ -91,34 +93,41 @@ void main() {
   });
 
   group('hotRestart', () {
-    final List<FlutterDevice> devices = <FlutterDevice>[];
+    final MockResidentCompiler residentCompiler = new MockResidentCompiler();
+    MockLocalEngineArtifacts mockArtifacts;
 
     setUp(() {
-      devices.add(new FlutterDevice(new MockDevice(),
-          previewDart2: false, trackWidgetCreation: false));
+      mockArtifacts = new MockLocalEngineArtifacts();
+      when(mockArtifacts.getArtifactPath(Artifact.flutterPatchedSdkPath)).thenReturn('some/path');
     });
 
     testUsingContext('no setup', () async {
-      expect((await new HotRunner(devices).restart(fullRestart: true)).isOk,
-          true);
+      final List<FlutterDevice> devices = <FlutterDevice>[new FlutterDevice(new MockDevice(), generator: residentCompiler, trackWidgetCreation: false, previewDart2: true)];
+      expect((await new HotRunner(devices).restart(fullRestart: true)).isOk, true);
+    }, overrides: <Type, Generator>{
+      Artifacts: () => mockArtifacts,
     });
 
     testUsingContext('setup function succeeds', () async {
-      expect((await new HotRunner(devices).restart(fullRestart: true)).isOk,
-          true);
+      final List<FlutterDevice> devices = <FlutterDevice>[new FlutterDevice(new MockDevice(), generator: residentCompiler, trackWidgetCreation: false, previewDart2: true)];
```

Comment:
nit: These lines may be rip for a line break.

Reply:
I'm about to followup with another patch that deletes the previewDart2 parameter, which'll get it just barely short-enough that it'll fit reasonably on one slightly long line.